app/6502.c

```diff
@@ -1,175 +1,45 @@
 #include <stdio.h>
-#include <stdlib.h>
-#include <SDL2/SDL.h>
-#include "SDL_pixels.h"
-
-struct State
-{
-  int key_quit;
-  int key_up;
-  int key_left;
-  int key_right;
-  int key_down;
-};
-
-struct Point
-{
-  int x;
-  int y;
-};
-
-SDL_Window* init_window()
-{
-  // returns zero on success else non-zero
-  if (SDL_Init(SDL_INIT_EVERYTHING) != 0) {
-    printf("error initializing SDL: %s\n", SDL_GetError());
-  }
-
-  return SDL_CreateWindow("6502",
-      SDL_WINDOWPOS_CENTERED,
-      SDL_WINDOWPOS_CENTERED,
-      1000, 1000, 0); 
+#include "cpu.h"
+#include "sdl.h"
+
+void initSimulator (struct CPU * cpu, struct BUS * bus, int initial_address, char * programPath) {
```

Comment:
`initial_address` will always be controlled by us or chances that will received user controlled input?
Thinking about if some validations would be necessary or not.

Reply:
When we drift away from the easy6502 specification we can use the header to populate this, that's why I made as an argument to ease this change

Reply:
I would like to add more validations :)